libkbfs/fs_rpc.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2016 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"fmt"
+
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/client/go/protocol"
+	"golang.org/x/net/context"
+)
+
+type fsRPC struct {
+	config Config
+	log    logger.Logger
+}
+
+func (f fsRPC) favorites(ctx context.Context, path Path) (keybase1.ListResult, error) {
+	favs, err := f.config.KBFSOps().GetFavorites(ctx)
+	if err != nil {
+		return keybase1.ListResult{}, err
+	}
+	files := []keybase1.File{}
+	for _, fav := range favs {
+		if (fav.Public && path.Public) || (!fav.Public && !path.Public) {
+			files = append(files, keybase1.File{Name: fav.Name})
+		}
+	}
+	return keybase1.ListResult{Files: files}, nil
+}
+
+func (f fsRPC) tlf(ctx context.Context, path Path) (keybase1.ListResult, error) {
+	files := []keybase1.File{}
+
+	node, de, err := path.GetNode(ctx, f.config)
+	if err != nil {
+		return keybase1.ListResult{}, err
+	}
+
+	if node == nil {
+		return keybase1.ListResult{}, fmt.Errorf("Node not found for path: %s", path)
+	}
+
+	if de.Type == Dir {
+		children, err := f.config.KBFSOps().GetDirChildren(ctx, node)
+		if err != nil {
+			return keybase1.ListResult{}, err
+		}
+
+		// For entryInfo: for name, entryInfo := range children
+		for name := range children {
+			files = append(files, keybase1.File{Name: name})
```

Comment:
This only gives you the entry name, and not the full `/keybase/.../name` path.  Is that what you want?  The `keybase()`, and `root()` commands give the full path with a leading `/`.  I guess `favorites()` only gives the base name as well.  It should probably be consistent across all RPC types.